cmd/gke-gcloud-auth-plugin/cred/cred.go

```diff
@@ -97,16 +111,23 @@ type Options struct {
 func PrintCred(options *Options) error {
 	if options == nil {
 		options = &Options{
-			UseApplicationDefaultCredentials: false,
+			TokenProvider: Gcloud,
 		}
 	}
 
 	var tokenProvider tokenProvider = nil
-	if options.UseApplicationDefaultCredentials {
+	switch options.TokenProvider {
```

Comment:
nit: I would explicitly check gcloud, ADC, and Edge, then have a default case which should error out.

Reply:
Done, thanks